avaframe/ana3AIMEC/aimecTools.py

```diff
@@ -991,6 +1006,95 @@ def computeRunOut(cfgSetup, rasterTransfo, resAnalysisDF, transformedRasters, si
     return resAnalysisDF
 
 
+def computeRunoutLine(cfgSetup, rasterTransfo, transformedRasters, simRowHash, type, name='', basedOnMax=False):
+    """ compute the runout line as for each l coordinate the furthest affected s coordinate using the desired
+        resType and thresholdvalue, or if basedOnMax searching for the max value (used for reference line, point)
+        also add runout point based on max s value of runout line
+
+        Parameters
+        -----------
+        cfgSetup: configparser object
+            configuration settings, here: runoutResType and tresholdValue
+        rasterTransfo: dict
+            information on raster transformation
+        transformedRasters: dict
+            transformed rasters from simulation
+        simRowHash: hash
+            index of current simulation
+        type: str
+            options: simulation, line, point, poly
+        name: str
+            optional name to find raster in transfromedRasters dict
+        basedOnMax: bool
+            if not threshold of runoutResType is used but max value along s for each l
+
+        Returns
+        ---------
+        runoutLine : dict
+            coordinates of runout line in s, l, and x, y, index of rasterTransfo['s']mname and type
+            coordinates of runout point (max s extent) in s, l
+        """
+
+    # get info on coordinate grids
+    gridx = rasterTransfo['gridx']
+    gridy = rasterTransfo['gridy']
+
+    # fetch raster data
+    if name == '':
+        runoutResType = cfgSetup['runoutResType']
+        anaRaster = transformedRasters['newRaster' + runoutResType.upper()]
+        name = simRowHash
+    else:
+        anaRaster = transformedRasters[name]
+
+    # set to 0 where nans and flip order of rows to search for furthest point in s
+    anaRaster = np.where(np.isnan(anaRaster), 0, anaRaster)
+    anaRasterFlip = np.flip(anaRaster, axis=0)
+
+    # setup runout line
+    thresholdValue = cfgSetup.getfloat('thresholdValue')
+    runoutLine = {'s': np.zeros(rasterTransfo['l'].shape) * np.nan, 'l': np.zeros(rasterTransfo['l'].shape) * np.nan,
+                  'x': np.zeros(rasterTransfo['l'].shape) * np.nan, 'y': np.zeros(rasterTransfo['l'].shape) * np.nan,
+                  'runoutLineFound': np.zeros(rasterTransfo['l'].shape) * np.nan}
+    runoutLineIndex = np.zeros(rasterTransfo['l'].shape) * np.nan
+
+    # loop over each l coordinate to find max s coordinate
+    sMaxInd = np.nan
+    lMaxInd = np.nan
+    sMax = 0
+    for ind, lCoor in enumerate(rasterTransfo['l']):
+        index1 = False
+        if basedOnMax and (len(np.nonzero(anaRaster[:,ind])[0]) > 0):
+            index2 = np.argmax(anaRasterFlip[:,ind])
+            index1 = anaRaster.shape[0]-index2
+        elif (basedOnMax == False) and (len(np.nonzero(anaRaster[:,ind]> thresholdValue)[0]) > 0):
+            index1 = max(np.nonzero(anaRaster[:,ind] > thresholdValue)[0])
+        if index1 != False:
```

Comment:
no it is false but the if a point is found to for an l coordinate then index refers to an index of the array

Reply:
Oh, I think this is rather confusing (and maybe not the best practice). Since index1 is initialized as boolean, I suggest to stick with this and use a separate array. 
And looking at your code, index1 is never going to be False?! It might be NAN, but not false, at least as far as I can tell...